app/org/sagebionetworks/bridge/upload/GenericUploadFormatHandler.java

```diff
@@ -0,0 +1,217 @@
+
+import org.sagebionetworks.bridge.json.BridgeObjectMapper;
+import org.sagebionetworks.bridge.json.DateUtils;
+import org.sagebionetworks.bridge.json.JsonUtils;
+import org.sagebionetworks.bridge.models.GuidCreatedOnVersionHolderImpl;
+import org.sagebionetworks.bridge.models.healthdata.HealthDataRecord;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+import org.sagebionetworks.bridge.models.surveys.Survey;
+import org.sagebionetworks.bridge.models.upload.UploadFieldDefinition;
+import org.sagebionetworks.bridge.models.upload.UploadFieldType;
+import org.sagebionetworks.bridge.models.upload.UploadSchema;
+import org.sagebionetworks.bridge.services.SurveyService;
+import org.sagebionetworks.bridge.services.UploadSchemaService;
+
+/**
+ * <p>
+ * Processes uploads for the v2_generic format. This handler reads from
+ * {@link org.sagebionetworks.bridge.upload.UploadValidationContext#getUnzippedDataMap} and
+ * {@link org.sagebionetworks.bridge.upload.UploadValidationContext#getJsonDataMap} and updates the existing record in
+ * {@link org.sagebionetworks.bridge.upload.UploadValidationContext#getHealthDataRecord} and the attachment map in
+ * {@link org.sagebionetworks.bridge.upload.UploadValidationContext#getAttachmentsByFieldName}.
+ * </p>
+ * <p>
+ * There is some overlap between this code and IosSchemaValidationHandler. However, we have opted to build this handler
+ * as an entirely separate class rather than create a monolithic handler to do both. This is to isolate the v1_legacy
+ * stuff from the v2_generic stuff, and to avoid propagating the hacks that were created to address launch day data.
+ * Truly shared code has been refactored either into InitRecordHandler and UploadUtil.
+ * </p>
+ * <p>
+ * Name note: The upload format is generic, not the handler.
+ * </p>
+ */
+@Component
+public class GenericUploadFormatHandler implements UploadValidationHandler {
+
+    private SurveyService surveyService;
+    private UploadSchemaService uploadSchemaService;
+
+    /** Survey service, to get the survey if this upload is a survey. Configured by Spring. */
+    @Autowired
+    public final void setSurveyService(SurveyService surveyService) {
+        this.surveyService = surveyService;
+    }
+
+    /** Upload Schema Service, used to get the schema corresponding to the upload. This is configured by Spring. */
+    @Autowired
+    public final void setUploadSchemaService(UploadSchemaService uploadSchemaService) {
+        this.uploadSchemaService = uploadSchemaService;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void handle(@Nonnull UploadValidationContext context) throws UploadValidationException {
+        Map<String, byte[]> attachmentMap = context.getAttachmentsByFieldName();
+        Map<String, JsonNode> jsonDataMap = context.getJsonDataMap();
+        HealthDataRecord record = context.getHealthDataRecord();
+        ObjectNode dataMap = (ObjectNode) record.getData();
```

Comment:
Always an object node and not an array, string, etc?

Reply:
This is guaranteed by InitRecordHandler to always be an ObjectNode. Not sure why it's typed as a JsonNode in HealthDataRecord.

It might be worth refactoring so getData() is type ObjectNode instead of JsonNode, but that's a lot of files it would touch. Filed JIRA https://sagebionetworks.jira.com/browse/BRIDGE-1960 to track this refactor.